subprojects/diagnostics/src/main/java/org/gradle/api/plugins/ProjectReportsPlugin.java

```diff
@@ -40,9 +41,16 @@ public void apply(final Project project) {
         project.getPluginManager().apply(ReportingBasePlugin.class);
         @SuppressWarnings("deprecation")
         final ProjectReportsPluginConvention convention = project.getObjects().newInstance(org.gradle.api.plugins.internal.DefaultProjectReportsPluginConvention.class, project);
-        @SuppressWarnings("deprecation")
-        Convention projectConvention = project.getConvention();
-        projectConvention.getPlugins().put("projectReports", convention);
+
+        // TODO expose `projectReports` as extension
```

Comment:
:question: Shouldn't this been done before we add the deprecation given accessing it would trigger the warning?

Reply:
Right, I wanted to get over this change a bit too much. Let me see if I can create a separate PR for moving the reports plugin to extensions.

Reply:
The todo is obsolete. The convention object there is only used internally to declare defaults. We already deprecated the corresponding `ProjectReportsPluginConvention` interface with the recommendation to configure reporting tasks directly.